File: challenges/strivers/3_check_palindrome.py
# Given an integer N, return true if it is a palindrome else return false

def isPalindrome(num: int) -> int:
    out = 0
    temp_num = num
    while num > 0:
        digit = num % 10
        out = out * 10 + digit
        num //= 10
    print(out)
    return temp_num == out


if __name__ == "__main__":
    print(isPalindrome(121))

File: challenges/strivers/3_check_palindrome.go
package strivers

func CheckPalindrome(num int) bool {

	if num < 0 {
		return false
	}

	temp_num := num
	out := 0

	for num > 0 {
		digit := num % 10
		out = out * 10 + digit
		num /= 10
	}
	return out == temp_num
}
